src/openforms/submissions/tests/test_submission_completion.py

```diff
@@ -232,6 +237,65 @@ def test_form_auth_cleaned_after_completion(self):
         )
         self.assertNotEqual(value, "foo")
 
+    def test_user_defined_variables_set_properly(self):
+        form = FormFactory.create()
+        step = FormStepFactory.create(
+            form=form,
+            form_definition__configuration={
+                "components": [
+                    {
+                        "type": "textfield",
+                        "key": "testComponent",
+                    },
+                ]
+            },
+        )
+        FormLogicFactory.create(
+            form=form,
+            json_logic_trigger={
+                "==": [
+                    {"var": "testComponent"},
+                    "test",
+                ]
+            },
+            actions=[
+                {
+                    "variable": "userDefinedVar",
+                    "action": {
+                        "type": "variable",
+                        "value": {"+": [{"var": "userDefinedVar"}, 1]},
+                    },
+                }
+            ],
+        )
+        submission = SubmissionFactory.create(form=form)
+        SubmissionStepFactory.create(
+            submission=submission,
+            form_step=step,
+            data={"testComponent": "test"},
+        )
+        SubmissionValueVariableFactory.create(
+            form_variable__key="userDefinedVar",
+            form_variable__source=FormVariableSources.user_defined,
+            form_variable__data_type=FormVariableDataTypes.int,
+            submission=submission,
+            key="userDefinedVar",
+            value=0,
+        )
+
+        self._add_submission_to_session(submission)
+        endpoint = reverse("api:submission-complete", kwargs={"uuid": submission.uuid})
+
+        response = self.client.post(endpoint, {"privacy_policy_accepted": True})
```

Comment:
Not a big deal, but the `{"privacy_policy_accepted": True}` is new of 2.1.x

Reply:
ah yeah, I deleted the three new tests since they showed up as a cherrypick merge conflict but missed this one!